diff --git a/examples/stream_rtsp_demo/main/src/main.cpp b/examples/stream_rtsp_demo/main/src/main.cpp
index b5df785..d1cf5ac 100644
--- a/examples/stream_rtsp_demo/main/src/main.cpp
+++ b/examples/stream_rtsp_demo/main/src/main.cpp
@@ -38,8 +38,9 @@ int _main(int argc, char* argv[])
     log::info("Camera width:%d height:%d format:%s fps:%d buffer_num:%d", cam_w, cam_h, image::fmt_names[cam_fmt].c_str(), cam_fps, cam_buffer_num);
 
     camera::Camera cam = camera::Camera(cam_w, cam_h, cam_fmt, "", cam_fps, cam_buffer_num);
-    camera::Camera *cam2 = cam.add_channel(640, 480);
+    //camera::Camera *cam2 = cam.add_channel(640, 480);
     display::Display disp = display::Display();
+    camera::Camera *cam2 = cam.add_channel(disp.width(), disp.height());
     auto audio_recorder = audio::Recorder();
     rtsp::Rtsp rtsp = rtsp::Rtsp();
     rtsp.bind_camera(&cam);
@@ -94,7 +95,7 @@ int _main(int argc, char* argv[])
         disp.show(*img);
         delete img;
         uint64_t curr_ms = time::ticks_ms();
-        // log::info("loop use %lld ms\r\n", curr_ms - last_ms);
+        log::info("loop use %lld ms\r\n", curr_ms - last_ms);
         last_ms = curr_ms;
     }
 
